toolkit/metric/metrics.go

```diff
@@ -0,0 +1,40 @@
+package metric
+
+type CounterRef struct{}
```

Comment:
Should we consider making `CounterRef` an `interface`? Then we just need to implement the methods rather than intercept methods?

Reply:
There is a question: why don't we keep it consistent with the existing code? (like: /toolkit/trace/span)

+ 1. Add interception method for /toolkit/xxx in inst
+ 2. Implement interception method in inst

Reply:
OK, if keep consistent with existing tracing toolkit, it's fine to me.